sdk/core/Azure.Core.Experimental/src/Serialization/ISerializedMemberNameProvider.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Reflection;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Provides mappings for member names to serialized property names.
+    /// </summary>
+    public interface ISerializedMemberNameProvider
+    {
+        /// <summary>
+        /// Maps a member <see cref="MemberInfo.Name"/> to a serialized property name.
+        /// </summary>
+        /// <param name="memberInfo">The <see cref="MemberInfo"/> to map to a serialized property name.</param>
```

Comment:
Nit - drop `property` in the comments everywhere else too because it might not necessarily be

Reply:
"property" there refers to the JSON property name, which is always a "property". I was careful to only use it in that context.

Reply:
Sure, but this interface isn't tied to JSON.